File: frontend/src/api/analytics.ts
import request from './request'
import type {
  OverviewData,
  ContentTrendData,
  ContentTrendParams,
  TopContentData,
  TopContentParams,
  DistributionData,
  RecentCommentsData,
  RecentCommentsParams,
} from '@/types/analytics'

export const analyticsApi = {
  getOverview() {
    return request.get<OverviewData>('/analytics/overview')
  },
  getContentTrend(params: ContentTrendParams) {
    return request.get<ContentTrendData>('/analytics/content-trend', { params })
  },
  getTopContent(params: TopContentParams) {
    return request.get<TopContentData>('/analytics/top-content', { params })
  },
  getDistribution() {
    return request.get<DistributionData>('/analytics/distribution')
  },
  getRecentComments(params: RecentCommentsParams) {
    return request.get<RecentCommentsData>('/analytics/recent-comments', { params })
  },
}

File: backend/internal/router/analytics.go
package router

import (
	"novablog/internal/controller"

	"github.com/gin-gonic/gin"
)

// RegisterAnalyticsRoutes 注册工作台统计路由。
func RegisterAnalyticsRoutes(r *gin.RouterGroup, analyticsController *controller.AnalyticsController, authMiddleware gin.HandlerFunc) {
	analytics := r.Group("/analytics")
	analytics.Use(authMiddleware)
	{
		analytics.GET("/overview", analyticsController.GetOverview)
		analytics.GET("/content-trend", analyticsController.GetContentTrend)
		analytics.GET("/top-content", analyticsController.GetTopContent)
		analytics.GET("/distribution", analyticsController.GetDistribution)
		analytics.GET("/recent-comments", analyticsController.GetRecentComments)
	}
}
